de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ft/features/faultTreeNodes/FaultTreeNodeMoveFeature.java

```diff
@@ -29,37 +34,82 @@
  */
 public class FaultTreeNodeMoveFeature extends VirSatMoveShapeFeature {
 
+	private static final String TRUE = "true";
+	private static final String IS_COMMENT = "is-comment";
+	private static final String BELONGS_TO_FAULT_NODE = "Belongs-to-fault-node";
+	private static final String Y_POS_REL_TO_FAULT_NODE = "y-pos-rel-to-fault-node";
+	private static final String X_POS_REL_TO_FAULT_NODE = "x-pos-rel-to-fault-node";
+
 	/**
 	 * Standard constructor
 	 * @param fp the feature provider
 	 */
 	public FaultTreeNodeMoveFeature(IFeatureProvider fp) {
 		super(fp);
 	}
-	
+
 	@Override
 	public void moveShape(IMoveShapeContext context) {
 		ContainerShape containerShape = (ContainerShape) context.getShape();
-		
+
+		PictogramElement pe = context.getPictogramElement();
+		FaultTreeNode bean = (FaultTreeNode) getBusinessObjectForPictogramElement(pe);
+
 		Connection targetConnection = context.getTargetConnection();
 		if (targetConnection != null) {
 			Anchor start = targetConnection.getStart();
 			Anchor end = targetConnection.getEnd();
-			
+
 			Anchor inputAnchor = AnchorUtil.getFreeAnchors(containerShape, AnchorType.INPUT).get(0);
+
 			Anchor outputAnchor = AnchorUtil.getOutputAnchor(containerShape);
-			
+
 			ReconnectionContext reconnectionContext = new ReconnectionContext(targetConnection, start, inputAnchor, null);
 			reconnectionContext.setReconnectType(ReconnectionContext.RECONNECT_TARGET);
 			getFeatureProvider().getReconnectionFeature(reconnectionContext).reconnect(reconnectionContext);
-			
+
 			CreateConnectionContext createConnectionContext = new CreateConnectionContext();
 			createConnectionContext.setSourceAnchor(outputAnchor);
 			createConnectionContext.setTargetAnchor(end);
 			new PropagationCreateFeature(getFeatureProvider()).create(createConnectionContext);
 		}
-		
+
 		super.moveShape(context);
+		moveComments(containerShape, bean);
+	}
+
+	/**
+	 * @param containerShape to layout comments for
+	 * @param bean business object of pictogram
+	 */
+	private void moveComments(ContainerShape containerShape, FaultTreeNode bean) {
+		ContainerShape parent = containerShape.getContainer();
+
+		EList<Shape> children = parent.getChildren();
+		for (Object element : children) {
+			Shape shape = (Shape) element;
+			if (isComment(shape) && shapeBelongsToFaultNode(shape, bean)) {
+				int xPos = Integer.parseInt(Graphiti.getPeService().getPropertyValue(shape, X_POS_REL_TO_FAULT_NODE));
+				int yPos = Integer.parseInt(Graphiti.getPeService().getPropertyValue(shape, Y_POS_REL_TO_FAULT_NODE));
+				Graphiti.getGaService().setLocation(shape.getGraphicsAlgorithm(), containerShape.getGraphicsAlgorithm().getX() + xPos, containerShape.getGraphicsAlgorithm().getY() + yPos);
+			}
+		}
 	}
-	
+
+	private boolean shapeBelongsToFaultNode(Shape shape, FaultTreeNode bean) {

```

Comment:
There seems to be duplicate code for working with the comment properties.
Maybe we can move them to a CommentUtil or something like that?